skymp5-server/cpp/server_guest_lib/WorldState.cpp

```diff
@@ -581,15 +581,30 @@ bool WorldState::AttachEspmRecord(const espm::CombineBrowser& br,
     LocationalDataUtils::GetRot(locationalData),
     FormDesc::FromFormId(worldOrCell, espmFiles)
   };
+
+  MpChangeFormREFR* changeForm = nullptr;
   if (!isNpc) {
     form.reset(new MpObjectReference(formLocationalData,
                                      formCallbacksFactory(), baseId,
                                      typeStr.data(), primitiveBoundsDiv2));
   } else {
-    form.reset(
-      new MpActor(formLocationalData, formCallbacksFactory(), baseId));
+    MpActor* actor =
```

Comment:
Please move this logic to a better place. Consider lazy loading: create a separate method for that and use only on attempt to get/set factions.

Also, we need to respect NPC templates. see example in MpActor::GetRaceId